include/flang/Optimizer/Dialect/FIROps.h

```diff
@@ -0,0 +1,76 @@
+//===-- Optimizer/Dialect/FIROps.h - FIR operations -------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef OPTIMIZER_DIALECT_FIROPS_H
+#define OPTIMIZER_DIALECT_FIROPS_H
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/OpDefinition.h"
+#include "mlir/IR/OpImplementation.h"
+#include "mlir/IR/SymbolTable.h"
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/ADT/StringRef.h"
+
+using namespace mlir;
+
+namespace fir {
+
+class FirEndOp;
+class LoopOp;
+
+enum class CmpFPredicate {
+  FirstValidValue,
+  // Always false
+  AlwaysFalse = FirstValidValue,
+  // Ordered comparisons
+  OEQ,
+  OGT,
+  OGE,
+  OLT,
+  OLE,
+  ONE,
+  // Both ordered
+  ORD,
+  // Unordered comparisons
+  UEQ,
+  UGT,
+  UGE,
+  ULT,
+  ULE,
+  UNE,
+  // Any unordered
+  UNO,
+  // Always true
+  AlwaysTrue,
+  // Number of predicates.
+  NumPredicates
+};
+
+void buildCmpFOp(mlir::Builder *builder, mlir::OperationState &result,
+                 CmpFPredicate predicate, mlir::Value lhs, mlir::Value rhs);
+void buildCmpCOp(mlir::Builder *builder, mlir::OperationState &result,
+                 CmpFPredicate predicate, mlir::Value lhs, mlir::Value rhs);
+unsigned getCaseArgumentOffset(ArrayRef<mlir::Attribute> cases, unsigned dest);
```

Comment:
It came from `using namespace mlir;` This using directive is required because of the generated `FIROps.h.inc` that does not use llvm and mlir namespace qualifier. I have qualified the `ArrayRef` you have mentioned and lowered the using directive to only impact the `FIROps.h.inc`  include.

Reply:
MLIR code often has implicit using on the `llvm` namespace. It's pretty ubiquitous, in fact.